### DISCUSSION TITLE
Deployment Setup - What should be the MVP to deploy a new Acoupi device?

### DISCUSSION THREAD

**Discussion by audevuilli** [acoupi/acoupi#6] [General]
Hello @mbsantiago  

I am creating a new discussion inside the GitHub repo to exchange about the building blocks of acoupi. Below is a copy of some questions/answers that would be good to agree on. 

(1) MVP for deployment (ideally for us and early testers). Here I assume a minimum knowledge of raspberry pi and command line usage. I think the following would be the "easiest".

- Install Raspberry Pi OS through RPi installer. - Set timezone, hostname and password.
- Copy the acoupi application to the RPi using curl and the GitHub link to the repo.
- Modify the provided configuration template (yaml, ...) - Enter location (i.e., latitude and longitude), device id following naming convention #NameFunction-#Number-#Location. Examples of devices deployed in the QEOP. ACOUPIBAT-DEV10-QEOP, ACOUPIBAT-DEV20-QEOP, ACOUPIBAT-DEV30-QEOP, ...
- Modify the configuration template for database access and storage.
- Run a .sh script (i.e., acoupi_start.sh) that installs all the dependencies and scripts to run the application.

(2) Advance stage deployment.

- Install Raspberry Pi OS through RPi installer. - Set timezone, hostname and password.
- Copy the acoupi application to the RPi using curl and the GitHub link to the repo.
- Here, a local interface similar to the BirdNET-Pi interface could be set up to configure latitude, longitude, device id, database access and storage, and all other possible configurations.

Happy to discuss your ideas if you think other methods would work better. **Especially would be nice to discuss a MVP configuration template**. I am quite familiar with a config.yaml file, but understand it might not be the best. 😃

**Comment by mbsantiago** [comment]
Hello @audevuilli,

Thanks for sharing ideas on the `acoupi` deployment process!

I agree that the configuration template is a good starting point, especially since it's easy to use and modify. Additionally, the .sh script for installation sounds like a great way to streamline the process for users.

I suggest using the factory set serial number of each RPi as a serial number is also a good one, as it can help avoid any potential mapping issues between ID and device. I've added a function utils module to extract this information.

For the advanced stage, I think creating a simple command line interface (CLI) to configure the `acoupi` could be nice. This could simplify the process for users who are already comfortable using the command line, and we could store the configurations in the background to keep track of what configurations were used and when. Additionally, the CLI could be used to create new deployments too.

I also like the idea of having a relatively simple web interface to interact with the `acoupi` for users who are not familiar with the command line. This could make it easier for a wider range of users to access the `acoupi` and should not be too hard to implement.

Regarding the configuration template, I think having the latitude, longitude, sampling rate, start_recording, end_recording, and record_interval is enough to replicate the Olympic park behavior and modify it to someone else's needs. However, it might be difficult to have a single configuration file if we've designed the `acoupi` to be flexible, especially if someone decides to implement a new `ScheduleManager` or a different `Model`. However, for a fixed selection of `Model`, `ScheduleManager`, and other parameters, we could have a config file.

Thanks again for starting the discussion and sharing your ideas 😄 !

Cheers,
@mbsantiago
